diff --git a/node_modules/react-native/scripts/react_native_pods_utils/script_phases.sh b/node_modules/react-native/scripts/react_native_pods_utils/script_phases.sh
index 6c41ce1..c90834c 100755
--- a/node_modules/react-native/scripts/react_native_pods_utils/script_phases.sh
+++ b/node_modules/react-native/scripts/react_native_pods_utils/script_phases.sh
@@ -13,8 +13,9 @@ GENERATED_SCHEMA_FILE="$GENERATED_SRCS_DIR/schema.json"
 
 cd "$RCT_SCRIPT_RN_DIR"
 
-CODEGEN_REPO_PATH="$RCT_SCRIPT_RN_DIR/packages/react-native-codegen"
-CODEGEN_NPM_PATH="$RCT_SCRIPT_RN_DIR/../react-native-codegen"
+# Please dont do this stuff
+# CODEGEN_REPO_PATH="$RCT_SCRIPT_RN_DIR/packages/react-native-codegen"
+# CODEGEN_NPM_PATH="$RCT_SCRIPT_RN_DIR/../react-native-codegen"
 CODEGEN_CLI_PATH=""
 
 error () {
@@ -23,14 +24,15 @@ error () {
     exit 1
 }
 
+# Please dont do this stuff
 # Determine path to react-native-codegen
-if [ -d "$CODEGEN_REPO_PATH" ]; then
-    CODEGEN_CLI_PATH=$(cd "$CODEGEN_REPO_PATH" && pwd)
-elif [ -d "$CODEGEN_NPM_PATH" ]; then
-    CODEGEN_CLI_PATH=$(cd "$CODEGEN_NPM_PATH" && pwd)
-else
-    error "error: Could not determine react-native-codegen location in $CODEGEN_REPO_PATH or $CODEGEN_NPM_PATH. Try running 'yarn install' or 'npm install' in your project root."
-fi
+# if [ -d "$CODEGEN_REPO_PATH" ]; then
+#     CODEGEN_CLI_PATH=$(cd "$CODEGEN_REPO_PATH" && pwd)
+# elif [ -d "$CODEGEN_NPM_PATH" ]; then
+#     CODEGEN_CLI_PATH=$(cd "$CODEGEN_NPM_PATH" && pwd)
+# else
+#     error "error: Could not determine react-native-codegen location in $CODEGEN_REPO_PATH or $CODEGEN_NPM_PATH. Try running 'yarn install' or 'npm install' in your project root."
+# fi
 
 find_node () {
     NODE_BINARY="${NODE_BINARY:-$(command -v node || true)}"
@@ -78,6 +80,13 @@ generateCodegenSchemaFromJavaScript () {
         echo "$RCT_SCRIPT_JS_SRCS_DIR" >> "${SCRIPT_OUTPUT_FILE_0}" 2>&1
     fi
 
+    # FML, RN is full of hardcoded paths that are incorrect when using monorepos.
+    # Welp, guess time to rewrite some shit. If only this came "out of the box" ...
+    CODEGEN_CLI_PATH=$("$NODE_BINARY" --print "require('path').dirname(require.resolve('react-native-codegen/package.json'))")
+    if ! [ -d "$CODEGEN_CLI_PATH" ]; then
+        error "error: Could not determine react-native-codegen location. Try running 'yarn install' or 'npm install' in your project root."
+    fi
+
     # shellcheck disable=SC2086
     # $JS_SRCS not having double quotations is intentional
     "$NODE_BINARY" "$CODEGEN_CLI_PATH/lib/cli/combine/combine-js-to-schema-cli.js" "$GENERATED_SCHEMA_FILE" $JS_SRCS
