diff --git a/node_modules/expo/build/environment/getInstallationIdAsync.js b/node_modules/expo/build/environment/getInstallationIdAsync.js
index 75366e4..5c83839 100644
--- a/node_modules/expo/build/environment/getInstallationIdAsync.js
+++ b/node_modules/expo/build/environment/getInstallationIdAsync.js
@@ -1,5 +1,5 @@
 import * as Application from 'expo-application';
-import uuidv5 from 'uuid/v5';
+import { v5 as uuidv5 } from 'uuid';
 let installationId;
 const UUID_NAMESPACE = '29cc8a0d-747c-5f85-9ff9-f2f16636d963'; // uuidv5(0, "expo")
 export default async function getInstallationIdAsync() {
