diff --git a/node_modules/expo/android/build.gradle b/node_modules/expo/android/build.gradle
index f8995f0..a60c4e1 100644
--- a/node_modules/expo/android/build.gradle
+++ b/node_modules/expo/android/build.gradle
@@ -33,7 +33,7 @@ def getRNVersion() {
 ensureDependeciesWereEvaluated(project)
 
 group = 'host.exp.exponent'
-version = '48.0.10'
+version = '48.0.9'
 
 buildscript {
   def expoModulesCorePlugin = new File(project(":expo-modules-core").projectDir.absolutePath, "ExpoModulesCorePlugin.gradle")
@@ -104,7 +104,7 @@ android {
     minSdkVersion safeExtGet("minSdkVersion", 21)
     targetSdkVersion safeExtGet("targetSdkVersion", 33)
     versionCode 1
-    versionName "48.0.10"
+    versionName "48.0.9"
     consumerProguardFiles("proguard-rules.pro")
   }
   lintOptions {
diff --git a/node_modules/expo/build/environment/getInstallationIdAsync.js b/node_modules/expo/build/environment/getInstallationIdAsync.js
index 75366e4..5c83839 100644
--- a/node_modules/expo/build/environment/getInstallationIdAsync.js
+++ b/node_modules/expo/build/environment/getInstallationIdAsync.js
@@ -1,5 +1,5 @@
 import * as Application from 'expo-application';
-import uuidv5 from 'uuid/v5';
+import { v5 as uuidv5 } from 'uuid';
 let installationId;
 const UUID_NAMESPACE = '29cc8a0d-747c-5f85-9ff9-f2f16636d963'; // uuidv5(0, "expo")
 export default async function getInstallationIdAsync() {
diff --git a/node_modules/expo/bundledNativeModules.json b/node_modules/expo/bundledNativeModules.json
index 783f93d..6e064e4 100644
--- a/node_modules/expo/bundledNativeModules.json
+++ b/node_modules/expo/bundledNativeModules.json
@@ -33,7 +33,7 @@
   "expo-crypto": "~12.2.1",
   "expo-dev-client": "~2.1.6",
   "expo-device": "~5.2.1",
-  "expo-document-picker": "~11.2.2",
+  "expo-document-picker": "~11.2.1",
   "expo-face-detector": "~12.1.1",
   "expo-file-system": "~15.2.2",
   "expo-font": "~11.1.1",
@@ -76,13 +76,13 @@
   "expo-system-ui": "~2.2.1",
   "expo-task-manager": "~11.1.1",
   "expo-tracking-transparency": "~3.0.3",
-  "expo-updates": "~0.16.4",
+  "expo-updates": "~0.16.3",
   "expo-video-thumbnails": "~7.2.1",
   "expo-web-browser": "~12.1.1",
   "lottie-react-native": "5.1.4",
   "react": "18.2.0",
   "react-dom": "18.2.0",
-  "react-native": "0.71.6",
+  "react-native": "0.71.4",
   "react-native-web": "~0.18.10",
   "react-native-branch": "^5.4.0",
   "react-native-gesture-handler": "~2.9.0",
